go/vt/sqlparser/sql.y

```diff
@@ -1114,26 +1115,36 @@ equal_opt:
     $$ = string($1)
   }
 
+name_opt:
+  ID
+  {
+    $$ = string($1)
+  }
+| /* empty */
+  {
+    $$ = ""
+  }
+
```

Comment:
Change is good.
Some nits:
* Existing code always lists the empty rule first. No particular reason, but it's better to follow existing convention.
* We generally don't mention `empty`. But there's no harm. However, let's use `//` instead.
* Also, lower level rules are listed below higher level ones, which means this should be moved after `index_or_key`.